src/Units/Unit.java

```diff
@@ -91,37 +92,30 @@ public double getBaseDamage(UnitModel target, int range, boolean moved)
     // if we have no weapons, we can't hurt things
     if( weapons == null )
       return 0;
+    double maxDamage = 0;
```

Comment:
If we are now calculating the weapon that does the most damage (which is fine by me), then we should update the function comment to reflect that. It currently says we choose the first weapon that can do damage.